src/parser/resolver/callExpressions/AbstractCollector.ts

```diff
@@ -40,7 +45,7 @@ export abstract class AbstractCollector {
                 source?: string;
             }[] = formatCaptures(tree, publicAccessorsCaptures);
 
-            console.log("public accessors captures", accessorsTextCaptures);
+            dlog("public accessors captures" + accessorsTextCaptures.toString());
```

Comment:
Debuglog uses the same argument handling as console.log. Was there a strong reason to change this?
That question applies to all of these changes.

Reply:
Mh, it did actually not work to simply replace console.log with debuglog in any case, as debuglog expects a string as first parameter, and console.log was sometimes used with just an object passed as parameter, which does not work with debuglog.

Reply:
The implementation that is triggered in Node.js passes the arguments through the same code paths. That should therefore not be different.

![image](https://github.com/MaibornWolff/metric-gardener/assets/8822573/68e5a254-825d-4a44-aa71-c30875eb2c15)

Reply:
Mh, ok. I don't know. I just got the error from the image below when trying to build the project with just replacing `console.log` with the debug logger. According to the IDE, the latter is defined as 
`export type DebugLoggerFunction = (msg: string, ...param: unknown[]) => void;`
while console.log is has the type
`log(message?:` any, ...optionalParams: any[]): void;`
which also accepts other types than "string" as first parameter.

![grafik](https://github.com/MaibornWolff/metric-gardener/assets/34129499/3ee0cd99-10d6-497e-8df6-7034de7d9139)

Reply:
The types are no official documentation of the functionality and the type definition for debuglog is just wrong. Looking at the official Node.js documentation, it is actually not documented at all how things are printed. Opening a PR to Node.js to improve that would be nice. To do so, I suggest to check the actual implementation of the feature.

Type definitions are inevitable faulty. They are defined by humans and it will therefore for certain deviate from the actual implementation from time to time. A great example where the type definition is likely always going to be somewhat wrong is `JSON.stringify()`. It has some functionality that is just very hard to write a type definition for.

In case of type errors you may therefore not rely upon them being correct using TypeScript.

Reply:
Interessting. 
For now, I just restored the old parameter passing where it was not strictly required to satisfy the type system, which is the case mentioned above.